Source/WebKit/WebProcess/GPU/graphics/RemoteResourceCacheProxy.cpp

```diff
@@ -141,15 +141,11 @@ void RemoteResourceCacheProxy::recordDecomposedGlyphsUse(DecomposedGlyphs& decom
     }
 }
 
-void RemoteResourceCacheProxy::releaseNativeImage(RenderingResourceIdentifier renderingResourceIdentifier)
+void RemoteResourceCacheProxy::releaseRenderingResource(RenderingResourceIdentifier renderingResourceIdentifier)
 {
-    auto iterator = m_nativeImages.find(renderingResourceIdentifier);
-    RELEASE_ASSERT(iterator != m_nativeImages.end());
-
-    auto success = m_nativeImages.remove(iterator);
-    ASSERT_UNUSED(success, success);
-
-    m_remoteRenderingBackendProxy.releaseRemoteResource(renderingResourceIdentifier);
+    bool removed = m_nativeImages.remove(renderingResourceIdentifier) || m_decomposedGlyphs.remove(renderingResourceIdentifier);
```

Comment:
I kind of don't like looking through all of the `HashMap`s to find the one that contains the resource.

Trying to think of some clever way to continue to have separate functions for releasing. Would this work? Have

* `RenderingResource` be generic, and the type parameter can be `NativeImage`, `DecomposedGlyphs`, etc.
* `NativeImage` inherit from `RenderingResource<NativeImage>`, etc.
* `RemoteResourceCacheProxy` take a reference to the object to release in its `releaseRenderingResource` functions, instead of its ID
* `RenderingResource<T>::Observer::release` call `RemoteResourceCacheProxy::release` with the object to release

Reply:
Though I guess this is already how it works in `RemoteResourceCache`. If you want to file an issue for a followup to avoid this in both places, that's OK.

Reply:
I agree. Another solution is to refactor `QualifiedResourceHeap` such that it can be used by `RemoteResourceCacheProxy` as it is used right now by `RemoteResourceCache`.